address/constants.go

```diff
@@ -1,62 +1,87 @@
+	if err != nil {
+		panic(err)
+	}
 
-// Base32CharsetReverse is the reverse character set. It maps ASCII byte -> Base32Charset index on [0,31].
-var Base32CharsetReverse = [128]int8{
-	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
-	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
-	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
-	15, -1, 10, 17, 21, 20, 26, 30, 7, 5, -1, -1, -1, -1, -1, -1,
-	-1, 29, -1, 24, 13, 25, 9, 8, 23, -1, 18, 22, 31, 27, 19, -1,
-	1, 0, 3, 16, 11, 28, 12, 14, 6, 4, 2, -1, -1, -1, -1, -1,
-	-1, 29, -1, 24, 13, 25, 9, 8, 23, -1, 18, 22, 31, 27, 19, -1,
-	1, 0, 3, 16, 11, 28, 12, 14, 6, 4, 2, -1, -1, -1, -1, -1,
-}
+	NetworkAddress, err = NewActorAddress([]byte("filecoin"))
+	if err != nil {
+		panic(err)
+	}
 
-// Base32 is a basex instance using the Base32Charset.
-var Base32 = basex.NewAlphabet(Base32Charset)
+	StorageMarketAddress, err = NewActorAddress([]byte("storage"))
+	if err != nil {
+		panic(err)
+	}
+
+	PaymentBrokerAddress, err = NewActorAddress([]byte("payments"))
+	if err != nil {
+		panic(err)
+	}
+}
 
 var (
-	// TestAddress is an account with some initial funds in it
+	// TestAddress is an account with some initial funds in it.
 	TestAddress Address
-	// TestAddress2 is an account with some initial funds in it
+	// TestAddress2 is an account with some initial funds in it.
 	TestAddress2 Address
-	// NetworkAddress is the filecoin network
+
+	// NetworkAddress is the filecoin network.
 	NetworkAddress Address
-	// StorageMarketAddress is the hard-coded address of the filecoin storage market
+	// StorageMarketAddress is the hard-coded address of the filecoin storage market.
 	StorageMarketAddress Address
-	// PaymentBrokerAddress is the hard-coded address of the filecoin storage market
+	// PaymentBrokerAddress is the hard-coded address of the filecoin storage market.
 	PaymentBrokerAddress Address
 )
 
-func init() {
-	t := Hash([]byte("satoshi"))
-	TestAddress = NewMainnet(t)
+var (
+	// ErrUnknownNetwork is returned when encountering an unknown network in an address.
+	ErrUnknownNetwork = errors.New("unknown address network")
+
+	// ErrUnknownProtocol is returned when encountering an unknown protocol in an address.
+	ErrUnknownProtocol = errors.New("unknown address protocol")
+	// ErrInvalidPayload is returned when encountering an invalid address payload.
+	ErrInvalidPayload = errors.New("invalid address payload")
+	// ErrInvalidLength is returned when encountering an address of invalid length.
+	ErrInvalidLength = errors.New("invalid address length")
+	// ErrInvalidChecksum is returned when encountering an invalid address checksum.
+	ErrInvalidChecksum = errors.New("invalid address checksum")
+)
 
-	t = Hash([]byte("nakamoto"))
-	TestAddress2 = NewMainnet(t)
+// UndefAddressString is the string used to represent an empty address when encoded to a string.
+var UndefAddressString = "[UNDEFINED_ADDRESS]"
```

Comment:
I suggest consistently using "empty" (preferred) or "undef" to refer to the non-address.